trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/file/FileBasedCredentialsProviderModule.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.credentials.file;
+
+import com.google.inject.Binder;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class FileBasedCredentialsProviderModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(FileBasedCredentialsProviderConfig.class);
+        binder.bind(CredentialsProvider.class).to(FileBasedCredentialsProvider.class).in(Singleton.class);
```

Comment:
I normally use `in(Scopes.SINGLETON)`. Is this the same?

Reply:
Both achieve the same purpose. Its a matter of coding preference. I've changed it to Scopes.Singleton to keep it conssitent with the rest of the codebase